util: Log static plugins meta data and used style

hebasto opened the pull request:
This PR is a follow-up of https://github.com/bitcoin/bitcoin/pull/17826, and adds additional info about the imported static plugins and the used style to the `debug.log` I found useful for testing (e.g., with `QT_QPA_PLATFORM`, `QT_QPA_PLATFORMTHEME`, `QT_STYLE_OVERRIDE` variables) and debugging issues (e.g., https://github.com/bitcoin/bitcoin/pull/19716#issuecomment-674052881).

The excerpt from the log:
```
2020-11-15T18:41:45Z [main] Bitcoin Core version v0.20.99.0-f0b933f78 (release build)
2020-11-15T18:41:45Z [main] Qt 5.9.8 (static), plugin=xcb (static)
2020-11-15T18:41:45Z [main] Static plugins:
2020-11-15T18:41:45Z [main]  QXcbIntegrationPlugin, version 329992
2020-11-15T18:41:45Z [main] Style: fusion / QFusionStyle
...
```

hebasto commented:
cc @fanquake

fanquake commented:
Concept ACK

MarcoFalke commented:
Concept ACK

promag commented on src/qt/guiutil.cpp:
```diff
@@ -912,6 +914,10 @@ void LogQtInfo()
     const std::string plugin_link{"dynamic"};
 #endif
     LogPrintf("Qt %s (%s), plugin=%s (%s)\n", qVersion(), qt_link, QGuiApplication::platformName().toStdString(), plugin_link);
+    for (const QStaticPlugin p : QPluginLoader::staticPlugins()) {
```
Could print "No static plugins" or something.

promag replied:
nit, use &?

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/72#issuecomment-699642353).

promag reviewed (commented):
Code review ACK 528360ade9bb3f4902b632f6f755cf63b3397412.

Tested but with brew qt 5.15.1, so no static plugins are listed.

hebasto commented:
Updated 528360ade9bb3f4902b632f6f755cf63b3397412 -> b5397ad1fbcb92eb244731aa60f07b33d55be797 ([pr72.01](https://github.com/hebasto/gui/commits/pr72.01) -> [pr72.02](https://github.com/hebasto/gui/commits/pr72.02), [diff](https://github.com/hebasto/gui/compare/pr72.01..pr72.02)):
- addressed all @promag's comments

promag commented on src/qt/guiutil.cpp:
```diff
@@ -912,6 +914,16 @@ void LogQtInfo()
+        LogPrintf("No static plugins.\n");
+    } else {
+        for (const QStaticPlugin& p : static_plugins) {
+            LogPrintf("Static plugin: %s\n", QJsonDocument(p.metaData()).toJson(QJsonDocument::Compact).toStdString());
```
nit, could write "Static plugins:" just once before the loop.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/72#issuecomment-702326522).

promag reviewed (commented):
Code review ACK b5397ad1fbcb92eb244731aa60f07b33d55be797.

hebasto commented:
Updated b5397ad1fbcb92eb244731aa60f07b33d55be797 -> 052f3eb5a7909c198dbd478220cbda203ebf2714 ([pr72.02](https://github.com/hebasto/gui/commits/pr72.02) -> [pr72.03](https://github.com/hebasto/gui/commits/pr72.03), [diff](https://github.com/hebasto/gui/compare/pr72.02..pr72.03)):

- addressed all @promag's [comment](https://github.com/bitcoin-core/gui/pull/72#discussion_r498098960):
> nit, could write "Static plugins:" just once before the loop.

hebasto commented:
Rebased 052f3eb5a7909c198dbd478220cbda203ebf2714 -> c45ecd3c4adf0de27f9429fec75d40c4a00276bc ([pr72.03](https://github.com/hebasto/gui/commits/pr72.03) -> [pr72.04](https://github.com/hebasto/gui/commits/pr72.04)).

laanwj commented on src/qt/guiutil.cpp:
```diff
@@ -40,11 +40,13 @@
 #include <QFontDatabase>
 #include <QFontMetrics>
 #include <QGuiApplication>
+#include <QJsonDocument>
```
does this really need JSON document? we use it nowhere else, and is not really our log format anway

promag replied:
I guess we could just print `QJsonObject QStaticPlugin::rawMetaData()`.

hebasto replied:
> I guess we could just print `QJsonObject QStaticPlugin::rawMetaData()`.

It prints something like that:
```
2020-11-15T18:02:40Z [main] Static plugins:
2020-11-15T18:02:40Z [main]  QTMETADATA  qbjs\x01
```

hebasto replied:
@laanwj @promag Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/72#issuecomment-727617533).

laanwj commented:
Concept ACK but I don't really see the point of JSON here.

hebasto commented:
Rebased c45ecd3c4adf0de27f9429fec75d40c4a00276bc -> f0b933f78d98a560394de5e09df8a6472433aa33 ([pr72.04](https://github.com/hebasto/gui/commits/pr72.04) -> [pr72.05](https://github.com/hebasto/gui/commits/pr72.05)):

- removed JSON from the log
- rebased on top of the recent CI changes

hebasto commented:
Rebased 5fffd6aa8853c3534f421c89006d98966c88fde9 -> 957895c715f86feaa26c806e5fa8ebb77430a926 ([pr72.06](https://github.com/hebasto/gui/commits/pr72.06) -> [pr72.07](https://github.com/hebasto/gui/commits/pr72.07)) on top of the recent CI changes.

jarolrod reviewed (commented):
ACK 957895c715f86feaa26c806e5fa8ebb77430a926, Tested on macOS 11.1

Since the OP Screenshot shows the case ***with*** static plugins, I tested the case ***without*** static plugins (only dynamic)

<img width="524" alt="Screen Shot 2021-01-18 at 9 11 25 PM" src="https://user-images.githubusercontent.com/23396902/104979324-02ce2280-59d2-11eb-8392-2dd8f21e786f.png">

jonasschnelli reviewed (approved):
utACK 957895c715f86feaa26c806e5fa8ebb77430a926

laanwj commented:
Posthumous ACK
